ietf-optical-impairment-topology.tree

```diff
@@ -169,14 +176,20 @@ module: ietf-optical-impairment-topology
        +--ro roadm-path-impairments-id    uint32
        +--ro (impairment-type)?
           +--:(roadm-express-path)
-          |  +--ro roadm-express-path
+          |  +--ro roadm-express-path* []
```

Comment:
Given that frequency-range is added for roadm add/drop/express path, would it be better to use frequency-range as key  for the three lists? 

Moreover, would it be better to replace roadm-path-impairments-id with frequency-range as key to roadm-path-impairments, and remove the case statement for "impairment-type". 

Also, why keep repeating the "roadm-" prefixes for pmd, cd etc. since their parent already has the same prefix. 

As a result the tree would look like the following:

+--ro roadm-path-impairments* [lower-frequency, upper-frequency]
     +-- lower-frequency      -> ../frequency-range/lower-frequency
     +-- upper-frequency      -> ../frequency-range/upper-frequency
     +--ro frequency-range
      |  +--ro lower-frequency    frequency-thz
      |  +--ro upper-frequency    frequency-thz
     +--roadm-express-path
         +--ro pmd
         +--ro cd 
                ......
     +--roadm-add-path
         +--ro pmd 
         +--ro cd 
                ......
     +--roadm-drop-path
         +--ro pmd
         +--ro cd 
                ......

Reply:
The 3 list roadm-express/add/drop-path are not configured list and it is not mandatory to have a key in this case. This why we have chosen do not put the key.

 would it be better to replace roadm-path-impairments-id with frequency-range as key to roadm-path-impairments, and remove the case statement for "impairment-type"?
Why?? What is the advantage when the tree as it is is very clear and readable?

About the roadm added in front to the attributes, this is due to an explicit request: we have pmd of cd for other elements in the topology and we 'd like to distinguish one to the other explicitly.

Reply:
Translating my tree proposal, it is to say that for each frequency range (C, L etc.) what would be the optical impairment for add, drop or express path. Don't you think it is a clearer expression? :)

IMO the original proposal is adding an unnecessary impairment-id with case statement to differentiate add, drop and express path (which all have a redundant list of frequency ranges).  It ends up with double lists. I do not see the clarity there.

Having a key using frequency range is just a natural outcome of the adjustment of the tree. You can always remove the key if you wish. 

On the "roadm-" prefixes I do not have a strong opinion, but just from a programming perspective I'd wish to say, if you already have the leaves defined in a context it is not necessary to add extra prefixes to each and every leaf within the same context. Right now you already have two levels of parent nodes with the same context, i.e. roadm-path-impairments and roadm-add-path. If you generate an xpath to the leaf it will be "roadm-" everywhere.

Reply:
It is clear a matter of opinion, I do not see any added value, but we can discuss a lot of time without conclusion.
About the key, as already said. no need to add a key in case the list is not config.
For prefix, and I think the request was to have the possibility to indicate directly the attributes at belonging to one or the other element. There was agreement in the group to add the prefix, but also for me, not strong opinion about.

Reply:
No need to add a key, but maybe need to put some constraints in the range (maybe in the description?) so that no two element in the list have the same range or are overlapping, but with different attributes. Key would not be sufficient to ensure that. That said, I still prefer the current tree

Reply:
Aihua/Italo: The roadm-path-impairments a list which contains other lists for roadm-express-path, roadm-add-path and roadm-drop-path.
The outer roadm-path-impairments list requires a key since it is referenced by leafrefs in the connectivity matrix and LLCL, while the inner lists do not require any key since they are read-only and not referenced.
It may be worthwhile clarifying in the description that the frequency ranges of the inner lists should not overlap (as proposed by Esther).

Reply:
Addressed in https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/68/commits/b37adeb031de1b26a146fd899a7a2638fbbb6157

@agva123 @EstherLerouzic : please check and confirm the resolution